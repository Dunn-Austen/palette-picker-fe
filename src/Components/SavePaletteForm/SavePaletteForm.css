#palette-input {
  height: 18px;
}

.palette-form {
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 10vh;
  justify-content: space-around;
}
