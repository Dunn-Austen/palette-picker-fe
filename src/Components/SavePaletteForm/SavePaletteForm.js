import React, {useState, useEffect} from 'react';
import './SavePaletteForm.css';
import { postPalette } from '../../apiCalls';
import PropTypes from 'prop-types';


const SavePaletteForm = ({projects, updateProjects, paletteColors}) => {

  let [inputValue, setInputValue] = useState('');
  let [selectedProject, setCurrentProject] = useState({});
  let [inputError, setError] = useState('');
  let [projectError, setProjectError] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  }

  const handleProjectChange = (event) => {
    let optionID = Number(event.target.value);
    let currentProject = projects.find(project => optionID === project.id);
    setCurrentProject(currentProject);
  }

  const clickHandler = async () => {
    if (!inputValue) return setError('error')
    await setError('');
    if (!selectedProject.id) return setProjectError('Please select project');
    await setProjectError('');
    let newPalette = {
      title: inputValue,
      project_id: selectedProject.id,
      color_1_id: paletteColors.paletteColor_1,
      color_2_id: paletteColors.paletteColor_2,
      color_3_id: paletteColors.paletteColor_3,
      color_4_id: paletteColors.paletteColor_4,
      color_5_id: paletteColors.paletteColor_5,
    };
    await submitPalette(newPalette);
    await setInputValue('');
  }

  const submitPalette = async (palette) => {
    let newProjects = [...projects];
    await postPalette(palette)
      .then(palette => {
        newProjects.find(project => palette.project_id === project.id).palettes.push(palette);
      });
      updateProjects(newProjects)
  }

  const projectNames = projects.map((project) => {
    return (
  <option key={project.id} value={project.id}>Project {project.id}: {project.title}</option>
    )
  })

  return (
    <section className='form-section'>
      <form className='palette-form'>
        <select id="project-select" onChange={handleProjectChange}>
          <option>--Select Project--</option>
          {projectNames}
        </select>
        <div className='input-container'>
          <label htmlFor='palette-input'>Save palette in project: </label>
          <input className={inputError}
            id='palette-input'
            type='text'
            placeholder='Backsplash Option #1'
            onChange={handleInputChange}
            value={inputValue}
            />
        </div>
        <button
          type="button"
          className='post-palette'
          onClick={() => clickHandler()}>
          Save palette
        </button>
        {projectError && <p>{projectError}</p>}
      </form>
    </section>
  );
}

export default SavePaletteForm;

SavePaletteForm.propTypes = {
  projects: PropTypes.array,
  updateProjects: PropTypes.func,
  paletteColors: PropTypes.object
}
