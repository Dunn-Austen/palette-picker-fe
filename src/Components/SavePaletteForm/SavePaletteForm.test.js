import React from 'react';
import { shallow } from 'enzyme';
import SavePaletteForm from './SavePaletteForm';

describe('SavePaletteForm', () => {
  it('should match the snapshot', () => {
    const mockProjects = [
      {
        id: 1,
        title: 'kitchen'
      }
    ];
    const mockUpdateProjects = jest.fn();
    const mockPaletteColors = {
      paletteColor_1: '#1223456',
      paletteColor_2: '#1223456',
      paletteColor_3: '#1223456',
      paletteColor_4: '#1223456',
      paletteColor_5: '#1223456',
    };

    const wrapper = shallow(<SavePaletteForm 
      projects={mockProjects}
      updateProjects={mockUpdateProjects}
      paletteColors={mockPaletteColors}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
