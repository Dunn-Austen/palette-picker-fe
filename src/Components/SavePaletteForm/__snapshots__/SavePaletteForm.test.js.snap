// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavePaletteForm should match the snapshot 1`] = `
<section
  className="form-section"
>
  <form
    className="palette-form"
  >
    <select
      id="project-select"
      onChange={[Function]}
    >
      <option>
        --Select Project--
      </option>
      <option
        key="1"
        value={1}
      >
        Project 
        1
        : 
        kitchen
      </option>
    </select>
    <div
      className="input-container"
    >
      <label
        htmlFor="palette-input"
      >
        Save palette in project: 
      </label>
      <input
        className=""
        id="palette-input"
        onChange={[Function]}
        placeholder="Backsplash Option #1"
        type="text"
        value=""
      />
    </div>
    <button
      className="post-palette"
      onClick={[Function]}
      type="button"
    >
      Save palette
    </button>
  </form>
</section>
`;
